Assets/Scripts/Cannon.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using DG.Tweening;
+using UnityEngine;
+
+public class Cannon : MonoBehaviour
+{
+    private const float AimTime = 0.3f;
+    [SerializeField] private float _maxHorizontalAngle = 45f;
+    [SerializeField] private float _cooldown = 1.5f;
+    [SerializeField] private Projectile _projectilePrefab;
+    [SerializeField] private Transform _projectileSpawnPoint;
+    [SerializeField] private Transform _cannonModel;
+
+    private readonly List<Target> _activeTargets = new List<Target>();
+    private Target _chosenTarget;
+    private float _timer;
+
+    private void Awake()
+    {
+        _activeTargets.AddRange(FindObjectsOfType<Target>());
+    }
+
+    private void Update()
+    {
+        if (_timer < _cooldown)
+        {
+            _timer += Time.deltaTime;
+            return;
+        }
+
+        _timer = 0;
+
+        FindTargetAndShoot();
+    }
+
+    private void FindTargetAndShoot()
+    {
+        var availableTargets = _activeTargets.FindAll(CheckTargetAngleDelta);
```

Comment:
This line has two allocations:
1) method ground (`CheckTargetAngleDelta`) gets converted to a delegate
2) `FindAll` creates a new list